parsec/backend/asgi/rpc.py

```diff
@@ -75,3 +86,114 @@ async def anonymous_api(raw_organization_id: str):
+
+    # Reply to GET
+    if request.method == "GET":
+        if not organization_exists:
+            abort(404)
+        else:
+            return rpc_msgpack_rep({})
+
+    body = await request.get_data(cache=False)
+    try:
+        client_ctx = await verify_auth_request(body, request.headers, organization_id, backend)
+    except Exception:
+        abort(401)
+
+    # Unpack the verified body
+    if request.headers["Content-Type"] != "application/msgpack":
+        abort(400)
+    try:
+        msg = unpackb(body)
+    except SerdePackingError:
+        return rpc_msgpack_rep({"status": "invalid_msg_format"})
+
+    cmd = msg.get("cmd")
+    try:
+        if not isinstance(cmd, str):
+            raise KeyError()
+        cmd_func = backend.apis[ClientType.AUTHENTICATED][cmd]
+    except KeyError:
+        return rpc_msgpack_rep({"status": "unknown_command"})
+
+    rep = await cmd_func(client_ctx, msg)
+    return rpc_msgpack_rep(rep)
+
+
+async def verify_auth_request(
+    body: bytes, headers: Headers, org_id: OrganizationID, backend: BackendApp
+) -> AuthenticatedClientContext:
+    authorization_method = headers["Authorization"]
+    if authorization_method == "PARSEC-SIGN-ED25519":
+        return await verify_auth_request_ed25519(body, headers, org_id, backend)
+    else:
+        raise ValueError(f"Unexpected authorization method {authorization_method}")
+
+
+async def verify_auth_request_ed25519(
+    body: bytes, headers: Headers, org_id: OrganizationID, backend: BackendApp
+) -> AuthenticatedClientContext:
+    # Parse api version header.
+    raw_api_version = request.headers["Api-Version"]
+    [high_api_version, low_api_version] = raw_api_version.split(".")[0:2]
+    api_version = ApiVersion(version=high_api_version, revision=low_api_version)
+
+    # Retrieve & parse device_id from http-headers
+    raw_device_id = request.headers["Author"]
+    device_id = DeviceID(base64.decodebytes(raw_device_id).decode())
+
+    # Retrieve timestamp from http-headers
+    raw_timestamp = request.headers["Timestamp"]
+
+    # Retrieve & decode request signature
+    signature_bytes = base64.decodebytes(request.headers["Signature"])
+
+    try:
+        organization = await backend.organization.get(org_id)
+        user, device = await backend.user.get_user_with_device(org_id, device_id)
+    except (OrganizationNotFoundError, UserNotFoundError, KeyError) as exc:
+        raise ValueError("User not found: {}", exc)
+
+    if organization.is_expired:
+        raise ValueError("organization is dead :(")
+
+    if user.revoked_on and user.revoked_on <= pendulum.now():
+        raise ValueError("User is dead :(")
+
+    verify_key = device.verify_key()
+    if (
+        verify_key.verify(smessage=raw_device_id + raw_timestamp + body, signature=signature_bytes)
+        is None
+    ):
+        raise ValueError("Invalid signed message, can't auth :(")
+
+    client_ctx = AuthenticatedClientContext(
+        api_version=api_version,
```

Comment:
use `_settle_compatible_versions`

https://github.com/Scille/parsec-cloud/blob/b5da56f550a3325173aee55fb7239041735ed1ed/parsec/api/protocol/handshake.py#L269-L271

Reply:
done in 61086fb60